cg/services/orders/validate_order_services/validate_microbial_order.py

```diff
@@ -1,29 +1,34 @@
 from cg.exc import OrderError
-from cg.meta.orders.microbial_submitter import MicrobialSubmitter
+from cg.models.orders.constants import OrderType
 from cg.models.orders.order import OrderIn
 from cg.models.orders.samples import SarsCov2Sample
+from cg.services.orders.submitters.order_submitter import ValidateOrderService
 from cg.store.models import Customer
+from cg.store.store import Store
 
 
-class SarsCov2Submitter(MicrobialSubmitter):
-    """Class for validating Sars-Cov submissions."""
+class ValidateMicrobialOrderService(ValidateOrderService):
+
+    def __init__(self, status_db: Store):
+        self.status_db = status_db
 
     def validate_order(self, order: OrderIn) -> None:
-        """Validate order sample names."""
-        super().validate_order(order=order)
-        self._validate_sample_names_are_available(samples=order.samples, customer_id=order.customer)
+        if order.order_type == OrderType.SARS_COV_2:
+            self._validate_sample_names_are_available(
+                samples=order.samples, customer_id=order.customer
+            )
```

Comment:
Does this not warrant a `ValidateSarsCov2OrderService`? I feel like having an if-statement here defeats the purpose of the nice registry you set up earlier

Reply:
In the Improve-order-flow new backend validation services are made. I put it here with that in mind. So in the future the more "generic" validation services would be replaced and the current one would be phased out.